pkg/network/discovery.go

```diff
@@ -88,6 +90,11 @@ func (d *DefaultDiscovery) pushToPoolOrDrop(addr string) {
 
 // RequestRemote tries to establish a connection with n nodes.
 func (d *DefaultDiscovery) RequestRemote(n int) {
+	d.lock.RLock()
+	defer d.lock.RUnlock()
```

Comment:
Is it needed to perform send to `requestCh` while mutex is taken?

Reply:
Sure not, I was just lazy to put two lines here instead of one.